### PR TITLE
CASMCMS-7876: Update BOS cli for v2

### PR DESCRIPTION
**Pull request by rbak-hpe**
## Summary and Scope

This adds the BOS v2 api to the cli.  It also adds a new way to inject code into the generator so that it's possible to access data at a point that is not otherwise possible.  This is necessary in order to setup a new bulk update endpoint for the cli.

## Issues and Related PRs

* Resolves CASMCMS-7876

### Tested on:

  * Hela and others

### Test description:

We have been using this to call all the commands while testing BOS v2

## Risks and Mitigations

None

## Pull Request Checklist

- [X] Version number(s) incremented, if applicable
- [X] Copyrights updated
- [X] License file intact
- [X] Target branch correct
- [X] CHANGELOG.md updated
- [X] Testing is appropriate and complete, if applicable
- [X] [HPC Product Announcement](https://cray.slack.com/archives/C026TVCSXLH) prepared, if applicable

### CODE REVIEW

**Comment by erl-hpe** [comment]
This needs to have the lint failures in the build fixed.

**Review by erl-hpe — COMMENTED** [review]
Once the lint failures are fixed, this looks okay to me.

**Review by erl-hpe — APPROVED** [review]

File: cray/modules/bos/cli.py [hunk 1]
```diff
@@ -50,31 +73,104 @@
...
+        if not (filter_ids or filter_session):
+            raise Exception('Either the ids or session filter must be provided')
+        if filter_ids and filter_session:
+            raise Exception('Only one of the two filter options can be provided')
+        payload = {}
+        if filter_ids:
+            payload["filters"] = {"ids": filter_ids}
+        if filter_session:
+            payload["filters"] = {"session": filter_session}
+        if patch["value"]:
```
Comment (jsollom-hpe) [inline cray/modules/bos/cli.py:123]: Is the "value" key guaranteed to be here? Otherwise, we'll get a KeyError.

File: cray/modules/bos/cli.py [hunk 2]
```diff
@@ -50,31 +73,104 @@
...
+            payload["patch"] = json.loads(patch["value"])
+        else:
+            payload["patch"] = {}
+        if enabled["value"] is not None:
```
Comment (jsollom-hpe) [inline cray/modules/bos/cli.py:127]: Ditto here.

**Review by jsollom-hpe — APPROVED** [review]

File: cray/modules/bos/cli.py (hunk 1 above)
Comment (rbak-hpe) [reply cray/modules/bos/cli.py:123]: It's guaranteed.  patch is required and will be a dictionary with the keys "key" and "value".  Not sure why it's done this way, but it is.

File: cray/modules/bos/cli.py (hunk 2 above)
Comment (rbak-hpe) [reply cray/modules/bos/cli.py:127]: Ditto my last response

**Pull request merged**